x-pack/plugins/cases/public/components/all_cases/columns.tsx

```diff
@@ -51,35 +53,106 @@ const TagWrapper = styled(EuiBadgeGroup)`
+  showActions: boolean;
 }
-export const getCasesColumns = ({
-  actions,
+export const useCasesColumns = ({
   caseDetailsNavigation,
   filterStatus,
-  isModal,
+  handleIsLoading,
+  refreshCases,
+  showActions,
 }: GetCasesColumn): CasesColumns[] => {
-  const columns = [
+  const { loading: isLoadingCases, dispatchUpdateCaseProperty } = useGetCases();
+  // Delete case
+  const {
+    dispatchResetIsDeleted,
+    handleOnDeleteConfirm,
+    handleToggleModal,
+    isDeleted,
+    isDisplayConfirmDeleteModal,
+    isLoading: isDeleting,
+  } = useDeleteCases();
+
+  const [deleteThisCase, setDeleteThisCase] = useState<DeleteCase>({
+    id: '',
+    title: '',
+    type: null,
+  });
+  const confirmDeleteModal = useMemo(
+    () => (
+      <ConfirmDeleteCaseModal
+        caseTitle={deleteThisCase.title}
+        isModalVisible={isDisplayConfirmDeleteModal}
+        isPlural={false}
+        onCancel={handleToggleModal}
+        onConfirm={handleOnDeleteConfirm.bind(null, [deleteThisCase])}
+      />
+    ),
+    [deleteThisCase, isDisplayConfirmDeleteModal, handleToggleModal, handleOnDeleteConfirm]
+  );
+  const toggleDeleteModal = useCallback(
+    (deleteCase: Case) => {
+      handleToggleModal();
+      setDeleteThisCase({ id: deleteCase.id, title: deleteCase.title, type: deleteCase.type });
+    },
+    [handleToggleModal]
+  );
+
+  const handleDispatchUpdate = useCallback(
+    (args: Omit<UpdateCase, 'refetchCasesStatus'>) => {
+      dispatchUpdateCaseProperty({
+        ...args,
+        refetchCasesStatus: () => {
+          if (refreshCases != null) refreshCases();
+        },
+      });
+    },
+    [dispatchUpdateCaseProperty, refreshCases]
+  );
+
+  const actions = useMemo(
+    () =>
+      getActions({
+        deleteCaseOnClick: toggleDeleteModal,
+        dispatchUpdate: handleDispatchUpdate,
+      }),
+    [toggleDeleteModal, handleDispatchUpdate]
+  );
+
+  useEffect(() => {
+    handleIsLoading(isDeleting || isLoadingCases.indexOf('caseUpdate') > -1);
+  }, [handleIsLoading, isDeleting, isLoadingCases]);
+
+  useEffect(() => {
+    if (isDeleted) {
+      if (refreshCases != null) refreshCases();
+      dispatchResetIsDeleted();
+    }
+  }, [isDeleted, dispatchResetIsDeleted, refreshCases]);
+
+  return [
     {
       name: i18n.NAME,
       render: (theCase: Case | SubCase) => {
         if (theCase.id != null && theCase.title != null) {
-          const caseDetailsLinkComponent = !isModal ? (
-            <CaseDetailsLink
-              caseDetailsNavigation={caseDetailsNavigation}
-              detailName={isSubCase(theCase) ? theCase.caseParentId : theCase.id}
-              subCaseId={isSubCase(theCase) ? theCase.id : undefined}
-              title={theCase.title}
-            >
-              {theCase.title}
-            </CaseDetailsLink>
-          ) : (
-            <span>{theCase.title}</span>
-          );
+          const caseDetailsLinkComponent =
+            caseDetailsNavigation != null ? (
```

Comment:
Maybe just `caseDetailsNavigation ?` to handle any situation where it may be `undefined` as well

Reply:
`!= null` checks that the value is not `null` and that the value is not `undefined`

Reply:
Ahh sorry! Didn't realize it was single equals 👍🏾